app/controllers/users.js

```diff
@@ -60,11 +60,39 @@ exports.signin = (req, res, next) => {
   const user = req.body ? req.body : {};
   emailValidation(req.body.email);
   userService.signin(user).then((signedInUser) => {
-    const token = sessionManager.encode({ username: signedInUser.email });
+    sessionManager.generateTokenAccess(signedInUser).then((tokenAccess) => {
+      res.status(200);
+      res.set('renewId', tokenAccess.renewId);
+      res.set('expirationDate', tokenAccess.expirationDate);
+      res.set(sessionManager.HEADER_NAME, tokenAccess.token);
+      res.end();
+    });
+  }).catch((err) => {
+    next(err);
+  });
+};
+
+exports.signout = (req, res, next) => {
+  userService.signout(req.user).then((signedOutUser) => {
     res.status(200);
-    res.set(sessionManager.HEADER_NAME, token);
     res.end();
   }).catch((err) => {
     next(err);
   });
 };
+
+exports.renew = (req, res, next) => {
+  const user = req.user;
+  const token = req.token;
+  if (req.body.renew_id === token.renewId) {
+    sessionManager.generateTokenAccess(user).then((tokenAccess) => {
```

Comment:
Here we have a little issue, the `sessionManager.generateTokenAccess(user)` will generate us a brand new jwt, this means it will gonna recalculate the `maximumUsefulDate`, and that is not desired in this case, since we want to renew the `expirationDate` but keep the `maximuUsefulDate` fixed to the original value.